src/transformers/models/trillsson_efficientnet/feature_extraction_trillsson_efficientnet.py

```
@@ -0,0 +1,513 @@
+
+    @staticmethod
+    def _pad_for_rfft(input_tensor, fft_length, is_reverse=False):
+        """Pads `input_tensor` to `fft_length` on its inner-most `fft_rank` dims."""
+
+        fft_length = np.array([fft_length], dtype=np.int32)
+        fft_shape = [dim if dim != -1 else None for dim in fft_length]
+
+        # Edge case: skip empty tensors.
+        if input_tensor.ndim is not None and any(dim == 0 for dim in input_tensor.shape):
+            return input_tensor
+
+        # Slice the last FFT-rank dimensions from input_tensor's shape.
+        input_fft_shape = list(input_tensor.shape[-1:])
+
+        if input_tensor.ndim is not None and input_fft_shape is not None:
+            # In reverse, we only pad the inner-most dimension to fft_length / 2 + 1.
+            if is_reverse and fft_shape[-1] is not None:
+                fft_shape[-1] = fft_shape[-1] // 2 + 1
+
+            paddings = [[0, max(fft_dim - input_dim, 0)] for fft_dim, input_dim in zip(fft_shape, input_fft_shape)]
+
+            if any(pad > 0 for _, pad in paddings):
+                outer_paddings = [[0, 0]] * max((input_tensor.ndim - 1), 0)
+                input_tensor = np.pad(input_tensor, outer_paddings + paddings)
+
+            return input_tensor
+
+    def _mel_to_hertz(self, mel_values):
+        """Converts frequencies in `mel_values` from the mel scale to linear scale."""
+        return self.mel_break_frequency_hertz * (np.exp(mel_values / self.mel_high_frequency_q) - 1.0)
+
+    def _hertz_to_mel(self, frequencies_hertz):
+        """Converts frequencies in `frequencies_hertz` in Hertz to the mel scale."""
+        return self.mel_high_frequency_q * np.log(1.0 + (frequencies_hertz / self.mel_break_frequency_hertz))
+
+    def frame(self, signal, frame_length, frame_step, pad_end=False, pad_value=0, axis=-1):
+        """Expands `signal`'s `axis` dimension into frames of `frame_length`"""
+
+        assert signal.ndim >= 1, "Expect input waveform is not None"
```

Let's raise an error rather than adding an assert

---

Revise as suggestion